### PR TITLE
fix(adapter): Add Auction Price Adapter Audit Remediations

### PR DESCRIPTION
**Pull request by pblivin0x**
Remediations
+ Fix exponential adapter implementation ([Sherlock Issue 39](https://github.com/sherlock-audit/2023-06-Index-judging/issues/39))
+ Fix overflow checks in logarithmic and exponential adapters ([Sherlock Issue 1](https://github.com/sherlock-audit/2023-06-Index-judging/issues/1))
+ Change `scalingFactor` to 18 digits with precise multiplication in logarithmic and exponential adapters ([Sherlock Issue 42](https://github.com/sherlock-audit/2023-06-Index-judging/issues/42))

### CODE REVIEW

File: contracts/protocol/integration/auction-price/BoundedStepwiseLogarithmicPriceAdapter.sol [hunk 1]
```diff
@@ -55,7 +55,7 @@ contract BoundedStepwiseLogarithmicPriceAdapter {
         uint256 timeBucket = _timeElapsed / bucketSize;
 
         // Protect against logarithmic argument overflow
-        if (timeBucket > type(uint256).max / timeCoefficient) {
+        if (timeBucket > uint256(type(int256).max) / timeCoefficient) {
```
Comment (FlattestWhite) [inline contracts/protocol/integration/auction-price/BoundedStepwiseLogarithmicPriceAdapter.sol:58]: Add a test for the case that this code protects against?

File: contracts/protocol/integration/auction-price/BoundedStepwiseLogarithmicPriceAdapter.sol [hunk 2]
```diff
@@ -70,7 +70,7 @@ contract BoundedStepwiseLogarithmicPriceAdapter {
         if (lnExpression > type(uint256).max / scalingFactor) {
             return _getBoundaryPrice(isDecreasing, maxPrice, minPrice);
         }
-        uint256 priceChange = scalingFactor * lnExpression;
+        uint256 priceChange = FixedPointMathLib.mulWad(scalingFactor, lnExpression);
```
Comment (FlattestWhite) [inline contracts/protocol/integration/auction-price/BoundedStepwiseLogarithmicPriceAdapter.sol:73]: Add a test case for the case that this code protects against - i.e. moving from higher dp to lower dp.

**Review by FlattestWhite — COMMENTED** [review]
Add test case that tests that the fixes do actually demonstrate the fix

File: contracts/protocol/integration/auction-price/BoundedStepwiseLogarithmicPriceAdapter.sol (hunk 1 above)
Comment (pblivin0x) [reply contracts/protocol/integration/auction-price/BoundedStepwiseLogarithmicPriceAdapter.sol:58]: Yes thank you @FlattestWhite , will add in base branch before merging to main

File: contracts/protocol/integration/auction-price/BoundedStepwiseLogarithmicPriceAdapter.sol (hunk 2 above)
Comment (pblivin0x) [reply contracts/protocol/integration/auction-price/BoundedStepwiseLogarithmicPriceAdapter.sol:73]: Will add WETH -> USDC and USDC -> WETH examples

**Pull request merged**